app/core/Snaps/SnapsMethodMiddleware.ts

```diff
@@ -51,6 +65,26 @@ const snapMethodMiddlewareBuilder = (
       origin,
       RestrictedMethods.wallet_snap,
     ),
+    getSnap: controllerMessenger.call.bind(
+      controllerMessenger,
+      'SnapController:get',
+    ),
+    handleSnapRpcRequest: async (request: Omit<SnapRpcHookArgs, 'origin'>) => {
+      const snapId = getSnapIdFromRequest(request);
+
+      if (!snapId) {
+        throw new Error(
+          'snapMethodMiddlewareBuilder handleSnapRpcRequest: Invalid snap request: snapId not found',
+        );
+      }
+
+      return await handleSnapRequest(controllerMessenger, {
+        snapId: snapId as SnapId,
```

Comment:
```suggestion
        snapId,
```
https://github.com/MetaMask/metamask-mobile/pull/10829/files#r1765861509